src/SFML/Graphics/Shape.cpp

```diff
@@ -143,6 +144,48 @@ FloatRect Shape::getGlobalBounds() const
     return getTransform().transformRect(getLocalBounds());
 }
 
+Vector2f Shape::getGeometricCenter() const
+{
+    auto count = getPointCount();
+    if (count > 2)
+    {
+        const auto off  = getPoint(0);
+        const auto off2 = 2.f * off;
+        Vector2f   xy{};
+        float      twicearea{};
+
+        float f;
+
+        for (std::size_t i = 0, j = count - 1; i < count; j = i++)
+        {
+            const auto p1 = getPoint(i);
+            const auto p2 = getPoint(j);
+
+            f = (p1.x - off.x) * (p2.y - off.y) - (p2.x - off.x) * (p1.y - off.y);
+            twicearea += f;
+            xy += ((p1 + p2) - off2) * f;
+        }
+
+        f = twicearea * 3.f;
+
+        return xy / f + off;
+    }
+
+    if (count == 2) // a line
+    {
+        return (getPoint(0) + getPoint(1)) / 2.f;
+    }
+
+    if (count == 1) // a single point
+    {
+        return getPoint(0);
+    }
```

Comment:
Might this be better expressed as a `switch` statement?

Reply:
It's not okay to use that because it's a C++20 feature. Let's focus on readability and expressiveness first. Performance is a tertiary concern.

Reply:
Ok, so, I took a look at it using a `switch` and I didn't like it, but I'll change if you want it.